### ISSUE TITLE
Duplicate same-URL nuget.config entries make credential identity selection nondeterministic

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3447]
Split out of review discussion on #3428, where it was raised as blocking, examined, and agreed down to a follow-up.

### Behavior

`SourceResolver.ResolveConfiguredSources` accumulates sources into a `Dictionary<string, string>` keyed by source *name*:

```csharp
Dictionary<string, string> mergedSources = [];
...
foreach ((string name, string url) in mergedSources)
```

Nothing prevents two differently-named entries from carrying the same URL:

```xml
<packageSources>
  <clear />
  <add key="first"  value="https://feed.example/v3/index.json" />
  <add key="second" value="https://feed.example/v3/index.json" />
</packageSources>
<packageSourceCredentials>
  <first>  <add key="Username" value="user-a" /> <add key="ClearTextPassword" value="secret-a" /> </first>
  <second> <add key="Username" value="user-b" /> <add key="ClearTextPassword" value="secret-b" /> </second>
</packageSourceCredentials>
```

`NuGetSourceResolver.Match` selects among configured sources with `FirstOrDefault`, so which of the two identities is used depends on the enumeration order of that dictionary. `Dictionary<TKey, TValue>` enumeration order is not part of its contract, so the identity a request authenticates as is not guaranteed stable.

### Why this is not a leak, and not blocking

Both entries name the byte-identical URL, so whichever credential is chosen is one the user configured *for exactly the endpoint being requested*. Nothing travels to an endpoint it was not configured for. This is distinct from the trailing-slash mis-selection fixed in #3428, where two entries named *different* URLs and one entry's secret could be attached to the other's spelling.

It was also explicitly decided **not** to "fix" this by refusing to adopt a credential when the match is ambiguous. That would break authentication for a feed the user did credential — the exact failure #3417 was filed about.

### What is actually worth doing

The concern that survives is only about *identity*: on a feed where `user-a` and `user-b` have different permissions, the same command could authenticate as either. Reasonable options:

- Pick deterministically (for example, first in document order) and document it.
- Detect duplicate URLs with differing credentials and report it, since that config is almost certainly a mistake — this fits the repo's "keep failure visible" rule better than an arbitrary silent choice.

Doing nothing is also defensible; the config is degenerate. Filing so the nondeterminism is recorded rather than rediscovered.

### Credit

Raised by the MAI-Code reviewer during adversarial review of #3428, and narrowed to this in discussion with the GPT reviewer.
